pkg/storaged/utils.js

```diff
@@ -47,6 +47,88 @@ export function compare_versions(a, b) {
     return a_ints.length - b_ints.length;
 }
 
+export function parse_options(options) {
+    if (options)
+        return (options.split(",")
+                .map(function (s) { return s.trim() })
+                .filter(function (s) { return s != "" }));
+    else
+        return [];
+}
+
+export function unparse_options(split) {
+    return split.join(",");
+}
+
+export function extract_option(split, opt) {
+    const index = split.indexOf(opt);
+    if (index >= 0) {
+        split.splice(index, 1);
+        return true;
+    } else {
+        return false;
+    }
+}
+
+export function edit_crypto_config(block, modify) {
+    let old_config, new_config;
+
+    function commit() {
+        new_config[1]["track-parents"] = { t: 'b', v: true };
+        if (old_config)
+            return block.UpdateConfigurationItem(old_config, new_config, { });
+        else
+            return block.AddConfigurationItem(new_config, { });
+    }
+
+    return block.GetSecretConfiguration({}).then(
+        function (items) {
+            old_config = items.find(c => c[0] == "crypttab");
+            new_config = ["crypttab", old_config ? Object.assign({ }, old_config[1]) : { }];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19388-20230928-091656-60b44b52-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)